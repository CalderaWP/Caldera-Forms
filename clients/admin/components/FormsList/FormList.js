import React from 'react';
import PropTypes from 'prop-types'
import {Form} from "./Form";

export const FormList = (props) => {
	let isAlternate = true;
	const forms = Array.isArray(props.form) ? props.form : Object.values(props.forms);
	return (
		<table className="widefat fixed">
			<thead>
			<tr>
				<th>Form</th>
				<th
					style={{width: '5em', textAlign: 'center'}}>
					Entries
				</th>
			</tr>
			</thead>
			<tbody>

			{

				forms.map(form => {
					isAlternate = ! isAlternate;
					return (
						<Form
							key={form.ID}
							form={form}
							onFormUpdate={props.onFormUpdate}
							openEntryViewerForForm={() => {
								alert(21);
								props.openEntryViewerForForm(form.ID);
							}}
							isAlternate={isAlternate}
						/>

					);
				})
			}
			</tbody>
		</table>
	);
};


FormList.propTypes = {
	forms: PropTypes.oneOfType([
		PropTypes.array,
		PropTypes.object,
	]),
	onFormUpdate: PropTypes.func.isRequired,
	openEntryViewerForForm: PropTypes.func.isRequired
};


