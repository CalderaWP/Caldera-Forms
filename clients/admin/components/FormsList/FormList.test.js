import renderer from "react-test-renderer";
import {FormList} from "./FormList";
import React from 'react';
import {mount} from 'enzyme';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
Enzyme.configure({adapter: new Adapter()});

const formOne = {
	ID: 'cf1',
	name: 'Form One',
	editLink: '/foo/bar',

};

const formWithEntries = {
	ID: 'cf2',
	name: 'Form Two',
	editLink: '/foo/bar/2',
	entries: {
		count: 42
	}
}

describe('Form List component', () => {
	it('Lists forms', () => {
		const formList = renderer.create(
			<FormList
				forms={[formOne]}
				onFormUpdate={()=> {}}
				openEntryViewerForForm={()=> {}}
			/>
		);
		expect(formList.toJSON()).toMatchSnapshot();
	});

	it('Passes form ID to openEntryViewerForForm ', () => {
		let id = '';
		const component = mount(
			<FormList
				forms={[formWithEntries]}
				onFormUpdate={()=> {}}
				openEntryViewerForForm={(formID)=> {
					id = formID;
				}}
			/>
		);

		component.find( '.view-entry-button').simulate('click');
		expect(id).toEqual(formWithEntries.ID);
	});

	it('Uses alternate class every other time, starting with the second', () => {
		const component = mount(
			<FormList
				forms={[formOne,formWithEntries,{
					...formOne,
					ID: 'CF4'
				}]}
				onFormUpdate={()=> {}}
				openEntryViewerForForm={()=> {}}
			/>
		);

		expect(component.find( '.alternate').length).toBe(1);
	});

	it('Does not use alternate class if only one form', () => {
		const component = mount(
			<FormList
				forms={[formOne]}
				onFormUpdate={()=> {}}
				openEntryViewerForForm={()=> {}}
			/>
		);

		expect(component.find( '.alternate').length).toBe(0);
	});


});