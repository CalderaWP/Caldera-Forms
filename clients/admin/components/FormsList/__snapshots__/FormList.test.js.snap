// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form List component Lists forms 1`] = `
<table
  className="widefat fixed"
>
  <thead>
    <tr>
      <th>
        Form
      </th>
      <th
        style={
          Object {
            "textAlign": "center",
            "width": "5em",
          }
        }
      >
        Entries
      </th>
    </tr>
  </thead>
  <tbody>
    <tr
      className="form_entry_row"
      id="form_row_cf1"
    >
      <td
        className=""
      >
        <span
          className="cf-form-name-preview"
        >
          Form One
        </span>
        <span
          className="cf-form-view-shorcode"
        >
          <button
            className="button cf-form-shortcode-preview-button"
            onClick={[Function]}
          >
            Get Shortcode
          </button>
        </span>
        <span
          className="row-actions"
        >
          <span
            className="edit"
          >
            <a
              href="/foo/bar"
            >
              Edit
            </a>
          </span>
        </span>
      </td>
      <td
        className="entry_count_cf1"
        style={
          Object {
            "textAign": "center",
            "width": "4em",
          }
        }
      >
        Disabled
      </td>
    </tr>
  </tbody>
</table>
`;
